drivers/vmwarevsphere/cloudinit.go

```diff
@@ -0,0 +1,272 @@
+				Value: d.CloudInit,
+			})
+		} else {
+			if _, err := os.Stat(d.CloudInit); err == nil {
+				if value, err := ioutil.ReadFile(d.CloudInit); err == nil {
+					log.Infof("setting guestinfo.cloud-init.data to encoded content of %s\n", d.CloudInit)
+					encoded := base64.StdEncoding.EncodeToString(value)
+					opts = append(opts, &types.OptionValue{
+						Key:   "guestinfo.cloud-init.config.data",
+						Value: encoded,
+					})
+					opts = append(opts, &types.OptionValue{
+						Key:   "guestinfo.cloud-init.data.encoding",
+						Value: "base64",
+					})
+				}
+			}
+		}
+	}
+
+	return d.applyOpts(vm, opts)
+}
+
+func (d Driver) uploadCloudInitIso(vm *object.VirtualMachine, dc *object.Datacenter, dss *object.Datastore) error {
+	path, err := d.getFolder(vm)
+	if err != nil {
+		return err
+	}
+
+	dsurl, err := dss.URL(d.getCtx(), dc, filepath.Join(path, isoName))
+	if err != nil {
+		return err
+	}
+
+	p := soap.DefaultUpload
+	c, err := d.getSoapClient()
+	if err != nil {
+		return err
+	}
+
+	if err = c.Client.UploadFile(d.getCtx(), d.ResolveStorePath(filepath.Join(isoDir, isoName)), dsurl, &p); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d Driver) removeCloudInitIso(vm *object.VirtualMachine, dc *object.Datacenter, dss *object.Datastore) error {
+	c, err := d.getSoapClient()
+	if err != nil {
+		return err
+	}
+
+	path, err := d.getFolder(vm)
+	if err != nil {
+		return err
+	}
+
+	m := object.NewFileManager(c.Client)
+	task, err := m.DeleteDatastoreFile(d.getCtx(), dss.Path(filepath.Join(path, isoName)), dc)
+	if err != nil {
+		return err
+	}
+
+	if err = task.Wait(d.getCtx()); err != nil {
+		if types.IsFileNotFound(err) {
+			// already deleted, ignore error
+			return nil
+		}
+
+		return err
+	}
+
+	return nil
+}
+
+func (d Driver) createCloudInitIso() error {
+	//d.CloudConfig stat'ed and loaded in flag load.
+	sshkey, err := ioutil.ReadFile(d.publicSSHKeyPath())
+	if err != nil {
+		return err
+	}
+
+	userdatacontent, err := addSshUserToYaml(d.CloudConfig, d.SSHUser, string(sshkey))
+	if err != nil {
+		return err
+	}
+
+	perm := os.FileMode(0700)
+	isoDir := d.ResolveStorePath(isoDir)
+	dataDir := filepath.Join(isoDir, "data")
+	userdata := filepath.Join(dataDir, "user-data")
+	metadata := filepath.Join(dataDir, "meta-data")
+
+	err = os.MkdirAll(dataDir, perm)
+	if err != nil {
+		return err
+	}
+
+	writeYaml := fmt.Sprintf("#cloud-config\n%s", userdatacontent)
+	if err = ioutil.WriteFile(userdata, []byte(writeYaml), perm); err != nil {
+		return err
+	}
+
+	md := []byte(fmt.Sprintf("#local-hostname: %s\n", d.MachineName))
+	if err = ioutil.WriteFile(metadata, md, perm); err != nil {
+		return err
+	}
+
+	//making iso
+	blocksize := int64(2048)
+	diskImg := filepath.Join(isoDir, isoName)
+	iso, err := os.OpenFile(diskImg, os.O_CREATE|os.O_RDWR, perm)
+	if err != nil {
+		return err
+	}
+	defer iso.Close()
+
+	fs, err := iso9660.Create(iso, 0, 0, blocksize)
+	fs.Mkdir("/")
```

Comment:
move `fs.Mkdir("/")` after the err check

Reply:
this is fixed w/ lint check